core/src/main/java/org/elasticsearch/script/StoredScriptSource.java

```diff
@@ -266,8 +262,7 @@ public static StoredScriptSource parse(String lang, BytesReference content, XCon
                         //this is really for search templates, that need to be converted to json format
                         try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
                             builder.copyCurrentStructure(parser);
-                            return new StoredScriptSource(lang, builder.string(),
-                                Collections.singletonMap(Script.CONTENT_TYPE_OPTION, XContentType.JSON.mediaType()));
+                            return new StoredScriptSource(lang, builder.string(), Collections.emptyMap());
```

Comment:
ok I am totally fine with removing the check for now and reintroducing it later once we have two different code-paths.